crates/state/src/wl_state.rs

```diff
@@ -1183,6 +1232,30 @@ where
     }
 }
 
+impl<D, H> EmitEvents for TempWlState<'_, D, H>
```

Comment:
The `TempWlState` is only used for validation/simulation in `mempool_validate` and `prepare/process_proposal`. Do we need to support events in those cases too?

Reply:
ah, I guess not. I implemented `EmitEvents` on pretty much every type that also implemented `State`. I'll remove this impl